src/Compilers/CSharp/Portable/Lowering/LocalRewriter_NullChecking.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        internal BoundStatement RewriteNullChecking(ImmutableArray<BoundStatement> statements, ImmutableArray<LocalSymbol> locals)
```

Comment:
>ImmutableArray<BoundStatement> statements, ImmutableArray<LocalSymbol> locals [](start = 52, length = 77)

Consider passing in the `BoundBlock` rather than the parts, to avoid the need to allocate a `BoundBlock` when there are no null-checked parameters.